plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -7623,6 +7623,95 @@ public void testSystemTables()
     }
 
+    @Test
+    public void testExtraProperties()
+    {
+        String tableName = "create_table_with_multiple_extra_properties_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE %s (c1 integer) WITH (extra_properties = MAP(ARRAY['extra.property.one', 'extra.property.two'], ARRAY['one', 'two']))".formatted(tableName));
+
+        assertQuery(
+                "SELECT value FROM \"%s$properties\" where key in ('extra.property.one', 'extra.property.two')".formatted(tableName),
+                "VALUES ('one'), ('two')");
+
+        assertUpdate("DROP TABLE %s".formatted(tableName));
+    }
+
+    @Test
+    public void testReplaceTableExtraProperties()
+    {
+        String tableName = "replace_table_with_multiple_extra_properties_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE %s (c1 integer) WITH (extra_properties = MAP(ARRAY['extra.property.one', 'extra.property.two'], ARRAY['one', 'two']))".formatted(tableName));
+        assertUpdate("CREATE OR REPLACE TABLE %s (c1 integer) WITH (extra_properties = MAP(ARRAY['extra.property.three'], ARRAY['three']))".formatted(tableName));
+
+        assertQuery(
+                "SELECT value FROM \"%s$properties\" where key in ('extra.property.one', 'extra.property.two', 'extra.property.three')".formatted(tableName),
+                "VALUES 'three'");
+    }
+
+    @Test
+    public void testExtraPropertiesWithCtas()
+    {
+        String tableName = "create_table_ctas_with_multiple_extra_properties_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE %s (c1 integer) WITH (extra_properties = MAP(ARRAY['extra.property.one', 'extra.property.two'], ARRAY['one', 'two']))".formatted(tableName));
+
+        assertQuery(
+            "SELECT value FROM \"%s$properties\" where key in ('extra.property.one', 'extra.property.two')".formatted(tableName),
+            "VALUES ('one'), ('two')");
+
+        assertUpdate("DROP TABLE %s".formatted(tableName));
+    }
+
+    @Test
+    public void testShowCreateWithExtraProperties()
+    {
+        String tableName = "show_create_table_with_extra_prop_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE %s (c1 integer) WITH (extra_properties = MAP(ARRAY['extra.property.one', 'extra.property.two'], ARRAY['one', 'two']))".formatted(tableName));
+
+        assertThat((String) computeActual("SHOW CREATE TABLE %s".formatted(tableName)).getOnlyValue()).doesNotContain("extra_properties");
+
+        assertUpdate("DROP TABLE %s".formatted(tableName));
+    }
+
+    @Test
+    public void testDuplicateExtraProperties()
+    {
+        assertQueryFails(
+                "CREATE TABLE create_table_with_duplicate_extra_properties (c1 integer) WITH (extra_properties = MAP(ARRAY['extra.property', 'extra.property'], ARRAY['true', 'false']))",
+                "Invalid value for catalog 'iceberg' table property 'extra_properties': Cannot convert.*");
```

Comment:
What's the following message of `Cannot convert.*`? Does the error message contain the duplicated key?

Reply:
No, the exception is `Invalid value for the 'iceberg' catalog's table property 'extra_properties': Cannot convert [MAP(ARRAY['extra.property', 'extra.property'], ARRAY['true', 'false'])] to map(varchar, varchar)` and I am just following the Hive implementation.

Following the stack trace, it is possible to see this error: `Caused by: io.trino.spi.block.DuplicateMapKeyException: Duplicate map keys (extra.property) are not allowed.`